lms/templates/courseware/course_navigation.html

```diff
@@ -21,6 +21,10 @@
     settings.FEATURES.get('ENABLE_SPECIAL_EXAMS', False) and
     (course.enable_proctored_exams or course.enable_timed_exams)
     )
+
+show_instructor = True
+if masquerade is not UNDEFINED and masquerade is not None and masquerade.role == 'student':
```

Comment:
You can re-use the existing utility method for this and replace the whole if condition with the following. 
```
if is_masquerading_as_student(user, course_key):
    show_instructor_tab = False
```

https://github.com/edx/edx-platform/blob/151bd136667d27bf85256e382ed5b5ea0144e00f/lms/djangoapps/courseware/masquerade.py#L337